TTS/encoder/models/resnet.py

```diff
@@ -277,8 +277,18 @@ def compute_embedding(self, x, num_frames=250, num_eval=10, return_mean=True, l2
     def load_checkpoint(self, config: dict, checkpoint_path: str, eval: bool = False, use_cuda: bool = False):
         state = load_fsspec(checkpoint_path, map_location=torch.device("cpu"))
         self.load_state_dict(state["model"])
+        # load the criterion for emotion classification
```

Comment:
You can do the same thing as above just create different loss classes and create a loss factory and use the factory in both places.

Reply:
Or you can implement base encoder and inherited by the subclasses

Reply:
Done, let me know if it is enough